runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -71,13 +95,27 @@ pub enum Error {
     #[error("gas price too low")]
     #[sdk_error(code = 7)]
     GasPriceTooLow,
+
+    #[error("insufficient balance")]
+    #[sdk_error(code = 8)]
+    InsufficientBalance,
+
+    #[error("invalid denomination")]
+    #[sdk_error(code = 9)]
+    InvalidDenomination,
+
+    #[error("core gas error: {0}")]
+    #[sdk_error(code = 10)]
+    CoreGasError(CoreError),
```

Comment:
Should probably use the transparent modifier as this can be any core error:
```suggestion
    #[error("core: {0}")]
    #[sdk_error(transparent)]
    Core(#[from] CoreError),
```